Supply matrixstack to IRenderHandler where possible. Closes #6439

williewillus opened the pull request:
Sky renderhandlers are completely useless currently because you don't have the matrix stack representing the current transform.

This isn't very clean and I'd rather just blow up irenderhandler and split it up, but an RB has already been released, so hacks it is.

LexManos commented on src/main/java/net/minecraftforge/client/IRenderHandler.java:
```diff
@@ -19,14 +19,27 @@
+     * @param matrixStack The matrix stack. Can be null when called at sites where vanilla does not supply one.
+     */
+    @OnlyIn(Dist.CLIENT)
+    default void render(int time, float partialTicks, ClientWorld world, @Nullable MatrixStack matrixStack) {
```
1) This breaks the @FunctionalInterface on certian compilers IIRC. As it only works with one method.
2) Why remove the MC argument? I understand you can get it from Minecraft.getInstane, but having it passed in seems like better practice.

tterrag1098 replied:
It won't, plenty of stdlib functional interfaces have lots of default methods. Just look at Comparator.

LexManos commented on src/main/java/net/minecraftforge/client/IRenderHandler.java:
```diff
@@ -19,14 +19,27 @@
 
+import javax.annotation.Nullable;
+
+// TODO 1.16 Split this apart for the various uses instead of stuffing everything into one interface.
```
Why not just do that now and make this class as deprecated, and have the children extend it?

williewillus commented:
updated to address comments.
added three new subinterfaces of IRenderHandler, with the eventual goal of killing IRenderHandler completely. patched worldrenderer to do instanceof and call the more specific method if possible.

Note: WeatherRenderHandler is empty because there is no matrix stack there, so its signature matches that of IRenderHandler, which it inherits. Once IRenderHandler is deleted, its method will be moved down into WeatherRenderHandler.

williewillus committed:
Supply matrixstack to IRenderHandler where possible. Closes #6439

HellFirePvP commented:
Is there a chance we'd be seeing this for 1.15 - or is this estimated to only be included in 1.16+ going forward?
